python/ray/dag/input_node.py

```diff
@@ -315,3 +315,35 @@ def __getitem__(self, key: Union[int, str]) -> Any:
                 "Please only use int index or str as first-level key to "
                 "access fields of dag input."
             )
+
+
+class MultiInputNode(DAGNode):
+    """InputNode with multiple args/kwargs packed inside"""
+
+    def __init__(
+        self,
+        *args,
+        **kwargs,
+    ):
+        counter = 0
+        self.idx2arg = {}
+        self.arg2idx = {}
+        for arg in args:
+            self.idx2arg[counter] = arg
+            counter = counter + 1
+        for kwarg in kwargs:
+            self.idx2arg[counter] = kwarg
+            counter = counter + 1
+        for idx, arg in self.idx2arg.items():
+            self.arg2idx[arg] = idx
+
+        super().__init__([], {}, {}, other_args_to_resolve=None)
+
+    def get_arg(self, idx: int):
```

Comment:
These methods don't seem to be used.

Also doesn't seem like we need the idx2arg arg2idx maps? Since you are already tracking the kwarg_to_idx outside of this class. Alternatively, you can move kwarg_to_idx from CompiledDAGNode into this class.